src/diff.rs

```diff
@@ -50,49 +58,97 @@ impl Default for Diff {
     }
 }
 
+#[cfg_attr(feature = "serialize", derive(Serialize))]
 #[derive(Debug, PartialEq, Eq)]
 pub struct CreateFile(pub Path);
 
+#[cfg_attr(feature = "serialize", derive(Serialize))]
 #[derive(Debug, PartialEq, Eq)]
 pub struct DeleteFile(pub Path);
 
+#[cfg_attr(
+    feature = "serialize",
+    derive(Serialize),
+    serde(rename_all = "camelCase")
+)]
 #[derive(Debug, PartialEq, Eq)]
 pub struct MoveFile {
     pub old_path: Path,
     pub new_path: Path,
 }
 
+#[cfg_attr(
+    feature = "serialize",
+    derive(Serialize),
+    serde(rename_all = "camelCase")
+)]
 #[derive(Debug, PartialEq, Eq)]
 pub struct CopyFile {
     pub old_path: Path,
     pub new_path: Path,
 }
 
+#[cfg_attr(
+    feature = "serialize",
+    derive(Serialize),
+    serde(rename_all = "camelCase")
+)]
 #[derive(Debug, PartialEq, Eq)]
 pub struct ModifiedFile {
     pub path: Path,
     pub diff: FileDiff,
 }
 
 /// A set of changes belonging to one file.
+#[cfg_attr(
+    feature = "serialize",
+    derive(Serialize),
+    serde(tag = "type", rename_all = "camelCase")
+)]
 #[derive(Debug, PartialEq, Eq)]
 pub enum FileDiff {
     Binary,
-    Plain(Vec<Hunk>),
+    #[cfg_attr(feature = "serialize", serde(rename_all = "camelCase"))]
+    Plain {
+        hunks: Vec<Hunk>,
+    },
 }
 
 /// A set of line changes.
+#[cfg_attr(
+    feature = "serialize",
+    derive(Serialize),
+    serde(rename_all = "camelCase")
+)]
 #[derive(Debug, PartialEq, Eq)]
 pub struct Hunk {
     pub header: Line,
     pub lines: Vec<LineDiff>,
 }
 
 /// The content of a single line.
-pub type Line = Vec<u8>;
+#[derive(Debug, PartialEq, Eq)]
+pub struct Line(pub Vec<u8>);
```

Comment:
Why does it need to be pub? Thought it's an anti-pattern with newtypes.

Reply:
We need it for serialization, and since there's no validation as to what can be a `Line`, I didn't see it as a problem. However, it should perhaps be `pub(crate)`!